pyop2/codegen/c/inverse.c

```diff
@@ -10,14 +11,33 @@ static PetscScalar work_buffer[BUF_SIZE*BUF_SIZE];
 
 static void inverse(PetscScalar* __restrict__ Aout, const PetscScalar* __restrict__ A, PetscBLASInt N)
 {
+    PetscLogEvent  USER_EVENT_memcpy;
+    PetscLogEvent  USER_EVENT_getrf;
+    PetscLogEvent  USER_EVENT_getrs;
+    PetscClassId   classid;
+
+    PetscClassIdRegister("PyOP2InverseCallable",&classid);
+    PetscLogEventRegister("PyOP2InverseCallable_memcpy",classid,&USER_EVENT_memcpy);
+    PetscLogEventRegister("PyOP2InverseCallable_getrf",classid,&USER_EVENT_getrf);
+    PetscLogEventRegister("PyOP2InverseCallable_gerti",classid,&USER_EVENT_getrs);
```

Comment:
It's probably fine as it is, but we could wrap these in preprocessor macros so they are conditionally included. For example:
```C
#ifdef PYOP2_PROFILING
PetscClassIdRegister(...);
...
#endif
```
I think that this is very unlikely to be a performance problem though.

Reply:
Okay since you said we can do that in a separate PR I opened an issue for it in #648 and will resolve this comment.